application/forms/IcingaDependencyForm.php

```diff
@@ -212,18 +213,50 @@ protected function addObjectsElement()
 
         if (!empty($sentParent) || $dependency->isApplyRule()) {
             $parentService = $dependency->get('parent_service');
+            if ($parentService === null) {
+                $parentServiceVar = $dependency->get('parent_service_by_name');
+                if ($parentServiceVar) {
+                    $parentService = '$' . $parentServiceVar . '$';
+                }
+            }
+
+            $applyTo = $this->getSentOrObjectValue('apply_to');
+
+            $description = $this->translate(
+                'Optional. The parent service. If omitted this dependency'
+                . ' object is treated as host dependency.'
+            );
+
+            if ($applyTo === 'service') {
+                $description .= $this->translate(
+                    ' You might want to refer to Service Custom Variables via $service.vars.varname$'
+                );
+            }
+
+            $callbackValidator = new Zend_Validate_Callback(function ($value) {
+                $applyTo = $this->getSentValue('apply_to');
+                if ($applyTo === 'service') {
+                    return true;
```

Comment:
In case of a service dependency, the user is able to enter `$host.vars.foo`. This may be fine, if it would be rendered as host custom variable, which it currently is not. Check if Icinga is able to resolve a host custom variable in case of a service dependency. If it works, don't render such as strings. If it doesn't, don't accept such as valid values.

Reply:
Icinga was able to resolve host custom variables. Hence, both host and service custom variable string patterns are rendered as custom variables.